### DISCUSSION TITLE
Proposed PR for OpenGraph Support for better integration with other websites

### DISCUSSION THREAD

**Discussion by danielfdickinson** [thegeeklab/hugo-geekdoc#205] [Ideas]
Hi,

I'm planning to add OpenGraph support to my GeekDoc-based website, and would like to know if there is interest by the developers (e.g. @xoxys) in my doing this in the style of the rest of the theme for upstreaming, or if this is something that I would maintain for myself only.

There are two parts to the idea. The first is the OpenGraph part which has no visible effect on the theme, it just adds metadata in the \<head> section (although with a .Site.Params.geekDocOpenGraphEnable / .page.Params.geekDocOpenGraphEnable check or such) using either the Hugo internal template for that, or a template based on the one from Hugo).

The advantage of OpenGraph is that forums like Discourse, as well as other sites, can use the supplied metadata and image (which can be a site-wide default, a per-page cover image is not required, but an image is required by the OpenGraph spec) to create a more appealing 'card' instead of 'just a link' to a page or article on one's site.

The other question is, if you have per-page images for OpenGraph, how (or whether) to display them (by default subject to an override or not by default by enabled by an override on some pages) on the page itself, without adding the image explicitly in the page markdown (or shortcode), or at least to have a shortcode with the same logic as the OpenGraph template for adding the image in a suitable place 'manually'.

The problem with images is how much effort to put in the rendering (e.g. responsive, converting format, sizing, etc).

So, I'm posting here for thoughts on the best way to approach this for GeekDoc (or to just do my own thing).

**Comment by xoxys** [comment]
Hi, there is already a opengraph integration available in Hugo, see e.g. https://github.com/thegeeklab/hugo-geekblog/blob/main/layouts/partials/head/microformats.html#L1

**Reply by danielfdickinson** [reply]
Doh! Thanks, I missed that... I guess it's not showing up because I haven't defined images for the template to find.

**Reply by xoxys** [reply]
Just to avoid confusion, the link above is another theme not `hugo-geekdoc` where the opengraph template is not included yet. So if you want to try it out, you would have to include the opengraph template on your own for now.

**Reply by danielfdickinson** [reply]
Ah ok, thank you for clarifying. I was aware of the template, what I meant (though obviously not stated clearly enough) was a proposal to add something like the microformats.html you showed, only as an opt-in (via ``. Param``) rather than always on. In any event, I'm happy enough to have as custom, just wondered if it were something that would be considered useful to add the theme.

**Reply by xoxys** [reply]
Sorry for the delay. I've stared to integrate some microformats in https://github.com/thegeeklab/hugo-geekdoc/pull/214. Be aware that you may have to provide the required metadata (title, description,...) to get it fully working.

**Reply by danielfdickinson** [reply]
This is awesome! Thank you.

**Comment by xoxys** [comment]
Official docs https://gohugo.io/templates/internal/#open-graph

**Comment by danielfdickinson** [comment]
Thanks again, sorry for the noise!

**Comment by danielfdickinson** [comment]
Thank you. Open Graph is working great for me with the microformats addition.
